### PR TITLE
Apply ps standards to CartRule class

### PR DESCRIPTION
**Pull request by firstred**
| Questions | Answers |
| --- | --- |
| Branch? | `develop` |
| Description? | Upgrading the PHPDoc comments of this class, using the latest PS standards and also applying the PS standards to some of the code. Also getting the Franglais out of the comments, getting rid of things the Symfony code sniffer nags about, etc. It might have some additional `todo` tags, depending on whether the code was comprehensible or not. |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | N/A |
| How to test? | N/A |

Warning: this PR also has a code cleanup. Please check this first before merging!

### CODE REVIEW

File: classes/CartRule.php [hunk 1]
```diff
@@ -699,202 +704,203 @@ public function checkValidity(Context $context, $alreadyInCart = false, $display
...
-            return (!$display_error) ? false : Tools::displayError('Cart is empty');
+        if (!$nbProducts) {
+            return (!$displayError) ? false : Tools::displayError('Cart is empty');
         }
 
-        if (!$display_error) {
+        if (!$displayError) {
             return true;
         }
     }
```
Comment (mickaelandrieu) [inline classes/CartRule.php]:
```
    /**
     * @param Context $context
     * @param bool $returnProducts
     * @param bool $displayError
     * @param bool $alreadyInCart
     * @return array|bool|string
     * @throws PrestaShopDatabaseException
     */
```

Comment (mickaelandrieu) [inline classes/CartRule.php]: sorry this comment is for the ``checkProductRestrictions`` function

Comment (firstred) [inline classes/CartRule.php]: Do you use PhpStorm?

Here are some settings to make it comply with Symfony standards:
phpdocsettings

Comment (mickaelandrieu) [inline classes/CartRule.php]: I use it yes, I'll take a look thanks !

**Pull request closed without merging**